server/app/services/applicant/ApplicantData.java

```diff
@@ -135,15 +199,19 @@ public void setUserName(String displayName) {
   }
 
   public void setUserName(
-      String firstName, @Nullable String middleName, @Nullable String lastName) {
+      String firstName, Optional<String> middleName, Optional<String> lastName) {
```

Comment:
comment (non-blocking): It's definitely not for you to address in this PR, but it's confusing to me to have a `setUserName` function and an `updateUserName` function. Maybe we can address this when we remove the WKP logic.